src/cc/tests/integtest_main.cc

```diff
@@ -1,11 +1,67 @@
 #include <gtest/gtest.h>
 #include <iostream>
+#include <string>
+#include <vector>
 
-#include "tests/environments/metaserver_environment.cc"
+#include <limits.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <unistd.h>
 
-GTEST_API_ int main(int argc, char **argv) {
-    std::cout << "Running main() from tests/integtest_main" << std::endl;
+#include "tests/environments/ChunkserverEnvironment.h"
+#include "tests/environments/MetaserverEnvironment.h"
+#include "tests/integtest.h"
+
+// TODO(fsareshwala): catch SIGINT and remove kill all environments and remove
+// all files
+
+using namespace KFS::Test;
+using namespace std;
+
+MetaserverEnvironment* QFSTest::sMetaserver = new MetaserverEnvironment;
+vector<ChunkserverEnvironment*> QFSTest::sChunkservers;
+const string QFSTestUtils::kTestHome = "/tmp/qfs";
+
+static void
+RandInit()
+{
+    struct timeval now;
+    gettimeofday(&now, NULL);
+
+    int pid = getpid();
+    int seed = (now.tv_sec * 1000 + now.tv_usec / 1000) ^ (pid * 0x5bd1e995);
+
+    srand(seed);
+    srandom(seed);
+}
+
+GTEST_API_ int
+main(int argc, char **argv)
+{
+    RandInit();
+
+    if (QFSTestUtils::FileExists(QFSTestUtils::kTestHome)) {
+        QFSTestUtils::RemoveForcefully(QFSTestUtils::kTestHome);
+    }
+
+    mkdir(QFSTestUtils::kTestHome.c_str(), S_IRWXU|S_IRWXG|S_IROTH|S_IXOTH);
+
+    cout << "Running custom main() from tests/integtest_main" << endl;
     testing::InitGoogleTest(&argc, argv);
-    testing::AddGlobalTestEnvironment(new KFS::Test::Metaserver);
-    return RUN_ALL_TESTS();
+    testing::AddGlobalTestEnvironment(QFSTest::sMetaserver);
+
+    static const int kNumChunkServers = 3;
+    for (unsigned int i = 0; i < kNumChunkServers; i++) {
+        ChunkserverEnvironment* env =
+            new ChunkserverEnvironment(QFSTest::sMetaserver);
+        QFSTest::sChunkservers.push_back(env);
+        testing::AddGlobalTestEnvironment(env);
+    }
```

Comment:
Sure, I think that might work. Let me make that change and submit another update.